services/platform-haproxy.service

```diff
@@ -25,15 +25,16 @@ ExecStartPre=/usr/bin/docker run -d --name fakedns --net protonet --net-alias so
 # Once haproxy is up, we need to get rid of the fakedns image because we want to actually resolve
 # to the actual apps once (and if they) become available
 ExecStartPost=-/usr/bin/docker rm -f fakedns
-ExecStartPre=/usr/bin/docker run -d \
+ExecStartPre=/usr/bin/env bash -c "/usr/bin/docker run -d \
     --net=protonet \
     --name platform-haproxy \
+    --env GITLAB_PUBLIC_IP=\"$(/opt/bin/gitlab-network show)\" \
```

Comment:
Please make sure that the gitlab network is already created and thus there's an actual ip address available by running after the appropriate service. Maybe even first only run the command and male sure it returns an ip address so that systemd error handling can apply?